open-craft/openedx-certificates
# feat: initial implementation [BB-7595]

**Merge checklist:**
Check off if complete *or* not applicable:

## Testing instructions
1. Use the `master` branch of the `edx-platform`.
2. Clone and install this package. Run migrations.
3. Upload the assets attached to the Jira ticket (private).
4. [Create](localhost:18000/admin/openedx_certificates/externalcertificatetype/add) two certificate types:
   1. Completion:
      1.  Retrieval func: `retrieve_course_completions`
      2. Generation func: `generate_pdf_certificate`
      3. Custom options:
           ```json
            {
                "name_y": 290,
                "course_name_y": 210,
                "template": "template_completion",
                "template_two_lines": "template_completion_two_line",
                "required_completion": 0.9
            }
            ```
   2. Achievement:
      1.  Retrieval func: `retrieve_subsection_grades`
      2. Generation func: `generate_pdf_certificate`
      3. Custom options:
           ```json
            {
                "name_y": 290,
                "course_name_y": 220,
                "template": "template_achievement",
                "required_grades": {
                    "Homework": 0.4,
                    "Exam": 0.9,
                    "Total": 0.8
                }
            }
            ```
5. [Create](localhost:18000/admin/openedx_certificates/externalcertificatecourseconfiguration/add) two course certificate configurations:
   1. Completion - you can use `course-v1:edX+DemoX+Demo_Course` as the course ID and override the `required_completion` to make the testing easier.
   2. Achievement - you can use `course-v1:edX+DemoX+Demo_Course` as the course ID and override the `required_grades` to make the testing easier.
6. Press the "generate certificates" button for each of the created course configurations.
7. [Check](http://localhost:18000/admin/openedx_certificates/externalcertificate/) that the certificates were generated correctly.

## Notes
1. The docs are deployed to https://openedx-certificates.readthedocs.io.
2. The pluggability of processors and generators will be included in the next iteration.

*Private-ref:* [BB-7595](https://tasks.opencraft.com/browse/BB-7595)
